docs/src/main/asciidoc/datastore.adoc

```diff
@@ -351,6 +351,59 @@ public class ConverterConfiguration {
 ----
 
 
+==== Inheritance Hierarchies
+
+Java entity types related by inheritance can be stored in the same Kind.
+When reading and querying entities using `DatastoreRepository` or `DatastoreTemplate` with a superclass as the type parameter, you can receive instances of subclasses if you annotate the superclass and its subclasses with `DiscriminationField` and `DiscriminationValue`:
+
+[source, java]
+----
+@Entity(name = "pets")
+@DiscriminationField(field = "pet_type")
+@DiscriminationValue("pet")
+abstract class Pet {
+	@Id
+	Long id;
+
+	abstract String speak();
+}
+
+@DiscriminationValue("cat")
+class Cat extends Pet {
+	@Override
+	String speak() {
+		return "meow";
+	}
+}
+
+@DiscriminationValue("dog")
+class Dog extends Pet {
+	@Override
+	String speak() {
+		return "woof";
+	}
+}
+
+@DiscriminationValue("pug")
+class Pug extends Dog {
+	@Override
+	String speak() {
+		return "woof woof";
+	}
+}
+----
+
+Instances of all 4 types are stored in the `pets` Kind.
```

Comment:
Pet is an abstract class, so only 3 types are stored in the kind.

Reply:
fixed !